### PR TITLE
Issue 26: Create CI for GitHub pages website

### LINKED ISSUE #26: Make `pkgdown` CI to auto-build and deploy website
We could also do with a new issue to make `pkgdown` CI to auto-build and deploy this.

_Originally posted by @seabbs in https://github.com/epinowcast/epidist/pull/19#pullrequestreview-2052629986_

### PR DESCRIPTION
**Pull request by athowes**
## Description

This PR will close #26.

I have run `usethis::use_pkgdown_github_pages()`. I get the error

```
Error in `gh::gh()`:
! GitHub API error (404): Not Found
✖ URL not found: <https://api.github.com/repos/epinowcast/epidist>
ℹ Read more at <https://docs.github.com/rest/repos/repos#update-a-repository>
Run `rlang::last_trace()` to see where the error occurred.
```

which I'm currently trying to fix (though could be a permission thing?).

## Checklist

- [ ] My PR is based on a package issue and I have explicitly linked it.
- [ ] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [ ] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [ ] I have tested my changes locally.
- [ ] I have added or updated unit tests where necessary.
- [ ] I have updated the documentation if required.
- [ ] My code follows the established coding standards.
- [ ] I have added a news item linked to this PR.
- [ ] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

File: .github/workflows/pkgdown.yaml [hunk 1]
```diff
@@ -1,13 +1,18 @@
-# Workflow derived from https://github.com/r-lib/actions/tree/master/examples
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
 # Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
 on:
   push:
-    branches: [main]
```
Comment (seabbs) [inline .github/workflows/pkgdown.yaml:5]: we don't want the master stuff here

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
+  pull_request:
+    branches: [main, master]
+  release:
+    types: [published]
   workflow_dispatch:
-  
+
 name: pkgdown
 
+permissions: read-all
```
Comment (seabbs) [inline .github/workflows/pkgdown.yaml:14]: not sure this is needed?

File: .github/workflows/pkgdown.yaml [hunk 3]
```diff
@@ -42,7 +49,7 @@ jobs:
 
       - name: Deploy to GitHub pages 🚀
         if: github.event_name != 'pull_request'
-        uses: JamesIves/github-pages-deploy-action@v4.6.0
+        uses: JamesIves/github-pages-deploy-action@v4.5.0
```
Comment (seabbs) [inline .github/workflows/pkgdown.yaml:52]: why role this back?

Comment (athowes) [reply .github/workflows/pkgdown.yaml:52]: I don't know. I used `usethis::use_pkgdown_github_pages()` and it automatically made these changes. Happy to revert this back forward.

**Comment by athowes** [comment]
Any ideas on what I should do (/if I should be aiming) to fix my API issue @seabbs?

File: .github/workflows/pkgdown.yaml (hunk 3 above)
Comment (seabbs) [reply .github/workflows/pkgdown.yaml:52]: yes please do.

**Comment by seabbs** [comment]
The build failure shows the issue with the current `pkgdown` deployment: https://github.com/epinowcast/epidist/actions/runs/9076507452/job/24939351635?pr=31#step:7:1

It looks like it is due to the `enwtheme` not being available which relates to your issue about it not having clear usage instructions. The fix is to use the approach that `epinowcast` takes (having it in the `DESCRIPTION` as part of the website dependencies but this also suggests that a small prr to `enwtheme` would really help others in the future.

**Comment by athowes** [comment]
I believe I have fixed these three issues (version, remove master, add theme dependency [I also added pkgdown, as listed on `epinowcast` `DESCRIPTION`]).

**Comment by athowes** [comment]
The build failed this time due to no `bookdown` package available: https://github.com/epinowcast/epidist/actions/runs/9084323528/job/24965006900?pr=31

I have added this package to Suggests. It would be nice if there were a way I could test this without fixing where it fails one by one using checks. Let me know if there is a better way I can do it.

**Comment by athowes** [comment]
This passed now, so perhaps can be reviewed / merged @seabbs.

**Review by seabbs — APPROVED** [review]
Lgtm. Nice one! Note we should start tracking issue resolution in the news file (see the pr checklist and epinowcast)

**Pull request merged**